lib/QuadTermUtil.ts

```diff
@@ -307,22 +307,32 @@ export function matchPatternComplete(quad: RDF.BaseQuad, pattern: RDF.BaseQuad):
  *
  * @param quad A quad - possibly containing variables
  * @param pattern A pattern - possibly containing variables
+ * @param options
+ *    skipVar - don't add variables in the quad to the mapping
+ *    returnMappings - return the mappings if it is a valid match
  */
-export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad): boolean {
-  const mapping: Record<string, RDF.Term> = {};
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad): boolean;
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings?: false }): boolean;
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings: true }): false | Record<string, RDF.Term>;
```

Comment:
Shouldn't this become the following?

```suggestion
export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings: true }): Record<string, RDF.Term>;
```

Reply:
No - as it returns false if there isn't a valid mapping (e.g. trying to map `ruben a person` to `?v ?v ?v`, we don't know if `?v` should be `ruben`, `a` or `person`)

Reply:
Ok, that makes sense.